packages/schema/compose/src/resolve.ts

```diff
@@ -961,13 +1080,8 @@ async function resolveLocalImports(
 
       const findImport = (
         def: GenericDefinition,
-        rootTypes: EnumOrObject[]
-      ) => {
-        // Skip objects that we've already processed
-        if (typesToImport[def.type]) {
-          return def;
-        }
-
+        rootTypes: EnumOrUnionOrObject[]
+      ): EnumOrUnionOrObject => {
```

Comment:
Changed this function to be more similar to its external imports's equivalent. Felt this function should indeed find the imported definition and return it. Instead it was finding the import, mutating the `typesToImport` object and then returning the `def` parameter.

This made sense given that the visitor should return the `def`, but it ties the intention of this function to its visitor consumer and also, I needed an actual findImport function to retrieve the union's memberTypes object defs and import them